packages/statusbar/src/statusbar.ts

```diff
@@ -42,16 +42,27 @@ export class StatusBar extends Widget implements IStatusBar {
    * @param statusItem - The item to add to the status bar.
    */
   registerStatusItem(id: string, statusItem: IStatusBar.IItem): IDisposable {
+    console.log(this._statusItems);
+    console.log(statusItem.item.title);
     if (id in this._statusItems) {
       throw new Error(`Status item ${id} already registered.`);
     }
 
+    if (
+      id === '@jupyterlab/application-extension:mode-switch' ||
+      id === '@jupyterlab/apputils-extension:kernel-status' ||
+      id === '@jupyterlab/codemirror-extension:line-col-status' ||
+      id === '@jupyterlab/notebook-extension:mode-status'
+    ) {
+      statusItem.priorityZoom = 1;
```

Comment:
Could you move this to the place the items are defined?

https://github.com/jupyterlab/jupyterlab/blob/5d7c44bdd89db72bf40ca7aeb8ecee92e07f7e06/packages/application-extension/src/index.tsx#L1183

https://github.com/jupyterlab/jupyterlab/blob/5d7c44bdd89db72bf40ca7aeb8ecee92e07f7e06/packages/apputils-extension/src/statusbarplugin.ts#L103

https://github.com/jupyterlab/jupyterlab/blob/5d7c44bdd89db72bf40ca7aeb8ecee92e07f7e06/packages/codemirror-extension/src/index.ts#L49

https://github.com/jupyterlab/jupyterlab/blob/5d7c44bdd89db72bf40ca7aeb8ecee92e07f7e06/packages/notebook-extension/src/index.ts#L431

Reply:
Hi @fcollonval just pushed up the changes you asked for.